elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,533 @@
+        headings.forEach(heading => {
+          // All buttons inside the `HowtoAccordionHeadings` are made
+          // unfocusable here. Only the first heading will be made focusable
+          // afterwards. This is necessary to implement roving tab index.
+          heading.querySelector('button').setAttribute('tabindex', -1);
+          const panel = this._panelForHeading(heading);
+
+          // Headings and panels need an ID so they can reference each other
+          // with the `aria-labelledby` and `aria-controls` attributes.
+          if (!heading.id)
+            heading.id = `howto-accordion-generated-${idCounter++}`;
+          if (!panel.id)
+            panel.id = `howto-accordion-generated-${idCounter++}`;
+          heading.setAttribute('aria-controls', panel.id);
+          panel.setAttribute('aria-labelledby', heading.id);
+
+          // Assign the appropriate roles to panels. Headings are custom
+          // elements and set their role in their own `connectedCallback`.
+          if (!panel.hasAttribute('role'))
+            panel.setAttribute('role', 'region');
+        });
+        // Make the first heading focusable.
+        headings[0].querySelector('button').setAttribute('tabindex', 0);
+
+        // Set all the panels to the collapsed state to have a well-defined
+        // initial state.
+
+        // Check if any of the headings have been marked as
+        // expanded using the `expanded` attribute. If so, all the associated
+        // panels get expanded as well.
+        headings
+          .forEach(heading => {
+            const panel = this._panelForHeading(heading);
+            if(!heading.expanded) {
+              this._collapseHeading(heading);
+              this._collapsePanel(panel);
+            } else {
+              this._expandHeading(heading);
+              this._expandPanel(panel);
+            }
+          });
+        });
+    }
+
+    /**
+     * `disconnectedCallback` removes all the event listeners that
+     * `connectedCallback` added.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('howto-change', this._onHowtoChange);
+      this.removeEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * `_isHeading` returns true if the given element
+     * is a `<howto-accordion-heading>`.
+     */
+    _isHeading(elem) {
+      return elem.tagName === 'HOWTO-ACCORDION-HEADING';
+    }
+
+    /**
+     * `_onHowtoChange` handles the custom `howto-change` event. The event’s
+     * target is the heading that has been instructed to expand (by click,
+     * keyboard input or by changing the `expanded` attribute or property).
+     */
+    _onHowtoChange(event) {
+      // If there’s an animation running, ignore the event.
+      if(this.classList.contains('animating')) return;
+      const heading = event.target;
+      const panel = this._panelForHeading(heading);
+      if(event.detail.isExpandedNow)
```

Comment:
I was convinced the google ESLint settings enforce something here, but apparently I was wrong. I’ll fix that in a separate PR as it’s gonna be big PR of just adding whitespace.